factory/processing/blockProcessorCreator.go

```diff
@@ -166,6 +166,13 @@ func (pcf *processComponentsFactory) newShardBlockProcessor(
 		return nil, err
 	}
 
+	if pcf.chainRunType == common.ChainRunTypeSovereign {
```

Comment:
Let's please make this clean without this if.

Reply:
added todo